File: src/main/java/me/leodev/trollingfreedomreborn/commands/TNTPlace.java
package me.leodev.trollingfreedomreborn.commands;

import org.bukkit.entity.EntityType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockPlaceEvent;

import java.util.ArrayList;

public class TNTPlace implements Listener {

    public static ArrayList<String> Fireball1 = new ArrayList();

    public void TNTPlace(Player p) {
        Player p2 = p.getPlayer();
        Fireball1.add(p2.getName());
    }

    public void UnTNTPlace(Player p) {
        Player p2 = p.getPlayer();
        Fireball1.remove(p2.getName());
    }

    @EventHandler
    public void onBlockPlace(BlockPlaceEvent e) {
        Player p = e.getPlayer();
        if (Fireball1.contains(p.getName())) {
            p.getLocation().getWorld().spawnEntity(e.getBlock().getLocation(), EntityType.TNT);
            e.getBlockPlaced().breakNaturally();
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/explosion/TNTPlace.kt
package com.leomadrassi.trollingfreedomreborn.trolls.explosion

import org.bukkit.entity.EntityType
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockPlaceEvent

class TNTPlace : Listener {

    companion object {
        val Fireball1 = ArrayList<String>()
    }

    fun TNTPlace(p: Player) {
        val p2 = p.player!!
        Fireball1.add(p2.name)
    }

    fun UnTNTPlace(p: Player) {
        val p2 = p.player!!
        Fireball1.remove(p2.name)
    }

    @EventHandler
    fun onBlockPlace(e: BlockPlaceEvent) {
        val p = e.player
        if (Fireball1.contains(p.name)) {
            p.location.world.spawnEntity(e.block.location, EntityType.TNT)
            e.blockPlaced.breakNaturally()
        }
    }
}
